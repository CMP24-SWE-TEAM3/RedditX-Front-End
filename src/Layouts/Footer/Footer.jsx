// imports
import React from 'react';
import {Container} from "./Footer.styled";
import { Link } from 'react-router-dom';

/**
 * Component that has a languages of different countries , privacy polices and content policies
 *
 * @returns {React.Component}
 */
const Footer = () => {
    return (
        <Container>
            <div className={'content'}>
                <div className={'menu-1'}>
                    <div className={'sec'}>
                        <Link href={"https://www.redditinc.com/policies/user-agreement"}>User Agreement </Link>
                        <Link href={"https://www.redditinc.com/policies/privacy-policy"}>Privacy Policy</Link>
                    </div>
                    <div className={'sec'}>
                        <Link href={"https://www.redditinc.com/policies/user-agreement"}>Content Policy </Link>
                        <Link href={"https://www.redditinc.com/policies/privacy-policy"}>Moderator Code of Conduct</Link>
                    </div>
                </div>
                <div className={'menu-1'}>
                    <div className={'sec'}>
                        <Link href={"https://www.redditinc.com/policies/user-agreement"}>English </Link>
                        <Link href={"https://www.redditinc.com/policies/privacy-policy"}>Français</Link>
                        <Link href={"https://www.redditinc.com/policies/privacy-policy"}>Italiano</Link>
                    </div>
                    <div className={'sec'}>
                        <Link href={"https://www.redditinc.com/policies/privacy-policy"}>Deutsch</Link>
                        <Link href={"https://www.redditinc.com/policies/user-agreement"}>Español</Link>
                        <Link href={"https://www.redditinc.com/policies/privacy-policy"}>Português</Link>
                    </div>
                </div>
                <div className={'inc'}>
                    Reddit Inc © 2022. All rights reserved
                </div>
            </div>
        </Container>
    );
};

export default Footer;
