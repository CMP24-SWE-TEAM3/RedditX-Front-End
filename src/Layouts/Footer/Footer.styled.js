import styled from "styled-components";

export const Container = styled.div`
  background-color: ${({theme}) => theme.background.primary};
  border: thin solid ${({theme}) => theme.lineColor.primary};
  border-radius: 4px;
  box-shadow: 0 2px 4px 0 rgb(0 0 0 / 6%);
  //width: 312px;
  height: 202px;
  margin-top: 16px;

  & .content {
    padding: 0;
    max-height: none;
  }

  & .menu-1 {
    border-bottom: 1px solid #e9ebed;
    display: flex;
    padding-top: 8px;
    padding-bottom: 8px;
    margin-left: 12px;
    margin-right: 12px;
  }

  & .sec {
    display: flex;
    flex-flow: column nowrap;
    width: 50%;
    padding: 0 4px;
    text-align: start;
  }

  & a {
    font-family: Noto Sans, Arial, sans-serif;
    font-size: 12px;
    font-weight: 400;
    color: ${({theme}) => theme.color.primary};
    line-height: 16px;
    display: inline-block;
    text-decoration: none;
    text-transform: capitalize;
    margin-top: 3px;
    margin-bottom: 3px;
  }

  & .inc {
    font-family: Noto Sans, Arial, sans-serif;
    font-size: 12px;
    font-weight: 400;
    color: ${({theme}) => theme.color.primary};
    line-height: 16px;
    display: flex;
    padding: 12px;
  }

  @media (max-width: 960px) {
    display: none;

  }
`
