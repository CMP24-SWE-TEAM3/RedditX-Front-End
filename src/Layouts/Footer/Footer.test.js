import {shallow} from 'enzyme';

// import Components
import Footer from "./Footer";

describe("Footer Component", () => {
    it("this is a test for Footer Component", () => {
        expect(shallow(<Footer/>)).toMatchSnapshot();

    });
});