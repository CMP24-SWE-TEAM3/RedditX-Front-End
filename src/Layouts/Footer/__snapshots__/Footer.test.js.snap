// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component this is a test for Footer Component 1`] = `ShallowWrapper {}`;
